app/static/src/app/utils.ts

```diff
@@ -208,3 +214,29 @@ export const runPlaceholderMessage = (selectedVariables: string[], sensitivity:
     const notRunYet = sensitivity ? userMessages.sensitivity.notRunYet : userMessages.run.notRunYet;
     return selectedVariables.length ? notRunYet : userMessages.model.noVariablesSelected;
 };
+
+export const convertAdvancedSettingsToOdin = (advancedSettings: AdvancedSettings) => {
+    const flattenedObject = Object.fromEntries(Object.entries(advancedSettings)
+        .map(([key, value]) => {
+            let numericVal: number | null;
+            if (value.standardForm) {
+                const firstValue = value.val[0] !== null ? value.val[0] : value.defaults[0];
+                const secondValue = value.val[1] !== null ? value.val[1] : value.defaults[1];
+                numericVal = firstValue * 10 ** secondValue;
+            } else {
+                numericVal = value.val !== null ? value.val : value.defaults;
+            }
+            return [key, numericVal];
+        })) as Record<AdvancedOptions, number>;
+
+    const advancedSettingsOdin: AdvancedSettingsOdin = {
+        atol: flattenedObject.Tolerance,
+        rtol: flattenedObject.Tolerance,
+        maxSteps: flattenedObject["Max steps"],
+        stepSizeMax: flattenedObject["Max step size"],
+        stepSizeMin: flattenedObject["Min step size"],
+        tcrit: flattenedObject["Critical time"]
```

Comment:
Not sure why using dot notation for Tolerance but array for everything else? Couldn't it use the enum values here?

Reply:
Tolerance is just a single word so the dot works but youre right, idk why i didnt just use the enum, done